internal/states/encryption/encryption.go

```diff
@@ -0,0 +1,263 @@
+package encryption
+
+import (
+	"fmt"
+	"github.com/hashicorp/go-hclog"
+	"github.com/opentofu/opentofu/internal/states/encryption/encryptionconfig"
+	"github.com/opentofu/opentofu/internal/states/encryption/encryptionflow"
+	"github.com/opentofu/opentofu/internal/tfdiags"
+	"sync"
+)
+
+// New creates a new Encryption object. You can use this object to feed in encryption configuration and then create
+// an encryptionflow.StateFlow or encryptionflow.PlanFlow as needed.
+//
+// Note: due to a lot of procedural code you probably want to use GetSingleton() to obtain the Encryption implementation
+// that's globally scoped.
+func New(logger hclog.Logger) Encryption {
+	return &encryption{
+		encryptionConfigs:         make(encryptionconfig.ConfigMap),
+		decryptionFallbackConfigs: make(encryptionconfig.ConfigMap),
+		mutex:                     &sync.Mutex{},
+		logger:                    logger,
+	}
+}
+
+// Encryption is the main interface to feed the encryption configuration and obtain the encryptionflow.Flow for running
+// the actual encryption.
+//
+// Note: due to a lot of procedural code you probably want to use GetSingleton() to obtain the Encryption implementation
+// that's globally scoped.
+type Encryption interface {
+	// ApplyEnvConfigurations applies a configuration coming from the operating system environment. The
+	// encryptionconfig.Config structures embedded in the parameters should have the source set to Environment in order
```

Comment:
It doesn't seem like the "source" is part of the encryptionconfig.Config structure anymore

Reply:
Yeah, seems like the method handles that now by itself.